insights/components/rhel_version.py

```diff
@@ -11,64 +11,48 @@
 In particular, an ``IsRhel*`` component can be added as a dependency of a
 parser to limit it to a given version.
 """
-
-from insights.core.plugins import component
 from insights.combiners.redhat_release import RedHatRelease
 from insights.core.dr import SkipComponent
+from insights.core.plugins import component
 
 
-@component(RedHatRelease)
-class IsRhel6(object):
+class IsRhel(object):
     """
-    This component uses ``RedHatRelease`` combiner
-    to determine RHEL version. It checks if RHEL6, if not
-    RHEL6 it raises ``SkipComponent``.
+    This component uses ``RedHatRelease`` combiner to determine the RHEL
+    major version. It then checks if the major version matches the version
+    argument, if it doesn't it raises ``SkipComponent``.
 
     Attributes:
-        minor (int): The minor version of RHEL 6.
+        minor (int): The minor version of RHEL.
 
     Raises:
-        SkipComponent: When RHEL version is not RHEL6.
+        SkipComponent: When RHEL major version does not match version.
     """
-    def __init__(self, rhel):
-        if rhel.major != 6:
-            raise SkipComponent('Not RHEL6')
+    def __init__(self, rhel, version=None):
+        if rhel.major != version:
+            raise SkipComponent("Not RHEL{vers}".format(vers=version))
         self.minor = rhel.minor
 
 
 @component(RedHatRelease)
-class IsRhel7(object):
-    """
-    This component uses ``RedHatRelease`` combiner
-    to determine RHEL version. It checks if RHEL7, if not \
-    RHEL7 it raises ``SkipComponent``.
```

Comment:
@ryan-blakley all of the child classes need to have docstrings so that they will be fully documented in the rendered documentation.

Reply:
@bfahr Ahh that's a good point I added them back.